gslib/commands/cp.py

```diff
@@ -605,7 +605,7 @@
 
                    gsutil -o "GSUtil:parallel_process_count=8" \\
                      -o "GSUtil:parallel_thread_count=1" \\
-                     -m cp -r /local/source/dir gs://bucket/path
+                     -m cp -j html -r /local/source/dir gs://bucket/path
```

Comment:
Not sure if this is a doc bug or an implementation bug, but with gsutil 4.38 `-j` will match the given pattern for any part of the filename, not just the extension. For example:

```
cball@cball:~/testo$ ls
a.txt  b.txt  txt.jpg

cball@cball:~/testo$ gsutil -o "GSUtil:parallel_process_count=8" -m cp -j txt -r ./ gs://do-not-delete-cball-test
Copying file://./a.txt [Content-Type=text/plain]...
Copying file://./b.txt [Content-Type=text/plain]...
Copying file://./txt.jpg [Content-Type=image/jpeg]...
/ [3/3 files][    0.0 B/    0.0 B]

cball@cball:~/testo$ gsutil ls gs://do-not-delete-cball-test
gs://do-not-delete-cball-test/a.txt
gs://do-not-delete-cball-test/b.txt
gs://do-not-delete-cball-test/txt.jpg
```

Should I open a bug for this and update the `-j` implementation, or is the doc inaccurate?

CC @houglum

Reply:
This is definitely something Matt should weigh in on, but how do you know that gzip transport encoding was applied to txt.jpg in your example?

Reply:
@starsandskies oops! I was confusing things, for some reason I was thinking that non-matches from `-j` would be excluded entirely from the copy. You're right that gzip encoding isn't applied to `txt.jpg` when running with `-D`. https://paste.googleplex.com/5962524765716480

Nevermind then! This PR looks good to me. 🙂

Reply:
Awesome! Glad we're good to go :D